internal/rpcserver/rpcserver.go

```diff
@@ -5920,7 +5920,8 @@ func (s *Server) route(ctx context.Context) *http.Server {
 
 		// Attempt to upgrade the connection to a websocket connection
 		// using the default size for read/write buffers.
-		ws, err := websocket.Upgrade(w, r, nil, 0, 0)
+		var upgrader websocket.Upgrader
```

Comment:
This needs to return true for the `CheckOrigin` function to avoid denying everything and keep the current behavior.  Also, since the `Upgrader` does not have an `Error` func defined, it will automatically handle the `http.Error` call, so that needs to be removed.

```diff
diff --git a/internal/rpcserver/rpcserver.go b/internal/rpcserver/rpcserver.go
index 867a96f2..7368a286 100644
--- a/internal/rpcserver/rpcserver.go
+++ b/internal/rpcserver/rpcserver.go
@@ -5913,15 +5913,16 @@ func (s *Server) route(ctx context.Context) *http.Server {

                // Attempt to upgrade the connection to a websocket connection
                // using the default size for read/write buffers.
-               var upgrader websocket.Upgrader
+               upgrader := websocket.Upgrader{
+                       CheckOrigin: func(r *http.Request) bool { return true },
+               }
                ws, err := upgrader.Upgrade(w, r, nil)
                if err != nil {
                        var herr websocket.HandshakeError
                        if !errors.As(err, &herr) {
-                               log.Errorf("Unexpected websocket error: %v",
-                                       err)
+                               log.Errorf("Unexpected websocket error: %v", err)
                        }
-                       http.Error(w, "400 Bad Request.", http.StatusBadRequest)
                        return
                }
```